trees/src/iavl/tree.rs

```diff
@@ -669,7 +872,7 @@ impl<'a, T: RangeBounds<Vec<u8>>, R: Database> Range<'a, T, R> {
                                 .get_node(&inner.right_hash)
                                 .expect("node db should contain all nodes");
 
-                            self.delayed_nodes.push(right_node);
+                            self.delayed_nodes.push(*right_node);
```

Comment:
Should `delayed_node` be a vec of boxed `Nodes` to avoid this copy?

Reply:
Changed to store boxes